tools/sgcloudspanner/emulator.go

```diff
@@ -15,17 +15,47 @@ import (
 	"go.einride.tech/sage/tools/sgdocker"
 )
 
+type Emulator struct {
+	ContainerID  string
+	DatabaseHost string
+	AdminHost    string
+	onClose      func()
```

Comment:
can we avoid this stateful private field?

Reply:
Yes, we should be able to deduce the close behavior from the containerID field on the struct.